app/com/arpnetworking/kairos/client/models/DataPoint.java

```diff
@@ -89,11 +76,11 @@ public String toString() {
 
     private DataPoint(final Builder builder) {
         _time = builder._time;
-        _value = builder._value;
+        _value = Optional.ofNullable(builder._value);
     }
 
     private final Instant _time;
-    private final Object _value;
+    private final Optional<Object> _value;
```

Comment:
From what [I've read in the past](https://blog.joda.org/2014/11/optional-in-java-se-8.html), the recommendation was to only use Optional.ofNullable in your accessor methods and store a nullable field.

So this would become
```suggestion
@Nullable
private final Object _value;
```

and the above
```java
    public Optional<Object> getValue() {
        return Optional.ofNullable(_value);
    }
```

Reply:
One of the main reasons for this recommendation is that Optional is not Serializable.  Since we use Jackson, we actually serialize as nullable in json and this won't bite us.  Using Optional like this is what we usually do, even though it's not Java standard.

Reply:
Makes sense. 

I think another point was that it could contribute to increased GC pressure if you have many short-lived objects with optional fields. But since neither of those cases apply here and theres precedent 🤷‍♂️

Reply:
To quote a colleague, "we should just use Kotlin".

Reply:
I wonder who could have said that? 🤔